crates/biome_js_formatter/src/ts/expressions/import_type_arguments.rs

```diff
@@ -0,0 +1,34 @@
+use crate::prelude::*;
+use biome_formatter::write;
+use biome_formatter::FormatError::SyntaxError;
+use biome_js_syntax::{TsImportTypeArguments, TsImportTypeArgumentsFields};
+#[derive(Debug, Clone, Default)]
+pub(crate) struct FormatTsImportTypeArguments;
+impl FormatNodeRule<TsImportTypeArguments> for FormatTsImportTypeArguments {
+    fn fmt_fields(&self, node: &TsImportTypeArguments, f: &mut JsFormatter) -> FormatResult<()> {
+        let TsImportTypeArgumentsFields {
+            l_paren_token,
+            argument,
+            comma_token,
+            ts_import_type_assertion_block,
+            r_paren_token,
+        } = node.as_fields();
+
+        if comma_token.is_some() && ts_import_type_assertion_block.is_some() {
+            write!(
+                f,
+                [
+                    l_paren_token.format(),
+                    argument.format(),
+                    comma_token.format(),
+                    space(),
+                    ts_import_type_assertion_block.format(),
+                    space(),
+                    r_paren_token.format()
+                ]
+            )
+        } else {
+            Err(SyntaxError)
+        }
```

Comment:
I think this is the code that caused the regression. When you have optional tokens, you can't emit a syntax error if they aren't present. In your case, you should do something like this:

```rs
fn main() {
            write!(
                f,
                [
                    l_paren_token.format(),
                    argument.format()
                ]
            )?;
	
	if let Some(comma_token) = comma_token {
		write!(f, [comma_token.format(), space()])?;
	}

	
	if let Some(ts_import_type_assertion_block) = ts_import_type_assertion_block {
		write!(f, [ts_import_type_assertion_block.format(), space()])?;
	}	

	write!(f, [r_paren_token.format()])
}
```

Also, **I strongly suggest you** to check how the JS import assertions are formatted. https://github.com/biomejs/biome/blob/main/crates/biome_js_formatter/src/js/module/import_assertion.rs

Reply:
wow, thank you here, i will take a look on it